source/functions/handle-errors.txt

```diff
@@ -0,0 +1,422 @@
+  Every time that the Function fails and enters the ``catch`` statement,
+  increment a count of the current number of retries.
+  Stop the recursive execution when the Function's current number of retries
+  reaches the max number of retries.
+- You may also want to `throttle <https://docs.microsoft.com/en-us/azure/architecture/patterns/throttling>`__
+  retries to reduce the total number of executions in a time frame.
+
+The following table describes some advantages and disadvantages
+of handling Function retries with recursive call strategy.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Advantages
+     - Disadvantages
+
+   * - - All retry logic occurs within one function.
+       - Function can return a value after retry.
+       - Minimal additional code.
+     - - All retries must occur within a single Function's max execution time.
+
+The following code example demonstrates an implementation of retrying a Function
+by using recursion in error handling blocks.
+
+.. literalinclude:: /examples/generated/functions/mightFail.snippet.recursive-error-handling.js
+   :language: js
+
+.. _functions-database-trigger-retry:
+
+Use Database Triggers to Retry
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also retry Functions by using a handler Function that manages retry attempts
+with a :ref:`Database Trigger <database-trigger>` to execute retries
+and a MongoDB collection to track previously failed executions.
+
+On a high-level, this process includes the following components:
+
+- **Main Function** that execute the logic you want to retry,
+  wrapped in the handler function (see below bullet point).
+- **Failed execution tracker collection**, a MongoDB collection
+  that tracks failed executions of the main Function.
+- **Handler Function** that invokes the main Function and logs when the function fails
+  to the failed execution tracker collection.
+- **Database Trigger Function** that reruns the handler function whenever
+  the handler function adds an error to the failed execution tracker collection.
+
+You can support multiple main functions with one set of a handler Function,
+execution tracker collection and Database Trigger Function.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 50 50
+
+   * - Advantages
+     - Disadvantages
+
+   * - - Each retry is its own Function execution, with own max execution time
+         and resources.
+     - - If the Function is retried, it cannot return a value.
+       - Each Function call requires two Function invocations, one for the Function
+         itself and one for the retry handler.
+       - More complex logic, which can be more difficult to write, debug, and monitor.
+
+.. procedure::
+
+   .. step:: Create a Function to handle execution retry
+
+      First, create a handler Function ``handleRetry`` that invokes the main Function.
+
+      ``handleRetry`` accepts the following parameters:
+
+      - ``functionToRetry``: the JavaScript function to retry.
```

Comment:
I wonder if these bullets would be better as a table, since the following bullets have effectively three columns - a name, a type, and a description.